src/internal/remote_descriptor/manager.cpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "internal/remote_descriptor/manager.hpp"
+
+#include "internal/remote_descriptor/remote_descriptor.hpp"
+#include "internal/remote_descriptor/storage.hpp"
+
+#include "srf/protos/codable.pb.h"
+
+namespace srf::internal::remote_descriptor {
+
+void Manager::decrement_tokens(std::size_t object_id, std::size_t token_count)
+{
+    DVLOG(10) << "decrementing " << token_count << " tokens from object_id: " << object_id;
```

Comment:
Is this thread safe, or is it assumed to be running on serial thread?

Reply:
Fixed with a mutex to protect the map.